### ISSUE TITLE
[FEA] Convert two columns into a new column as a dictionary

### ISSUE THREAD

**Issue by MikeChenfu** [rapidsai/cudf#8129]
**Is your feature request related to a problem? Please describe.**
Hello Guys,  I see `dict` in cudf is supported when I instantiate the df. `df = cudf.DataFrame({'a': [{'a':1},{'b':1}]})`.  Is there a method to convert two columns into a new column as a dictionary?

**Describe the solution you'd like**
```
df = cudf.DataFrame({'a': [1, 1, 1, 2, 2], 'b': [1, 1, 2, 2, 3], 'c': [1, 2, 3, 4, 5]})
df['d'] = {'a':df.a, 'b':df.b}

```

**Comment by kkraus14** [comment]
@MikeChenfu we're looking into this, but just to clarify, we don't support `map` types yet, which would be the equivalent of `dict` which allow arbitrary key-value pairs. We support `struct` types, which requires the same keys for every row of data.

**Comment by MikeChenfu** [comment]
Thanks @kkraus14 for the information.  Currently I am implementing hive `named_struct` using cudf. I am very interested in the `struct` type. Is cudf able to write it as orc file? Appreciate it if some docs or examples are provided.

**Comment by kkraus14** [comment]
> Thanks @kkraus14 for the information. Currently I am implementing hive `named_struct` using cudf. I am very interested in the `struct` type. Is cudf able to write it as orc file? Appreciate it if some docs or examples are provided.

We don't supporting writing struct columns to ORC quite yet, but it's being worked on currently. Hopefully it should land in the next release or two. Parquet writing is supported though.

**Comment by MikeChenfu** [comment]
Good to know that. Parquet is also good to me.:)

**Comment by beckernick** [comment]
The core request here has been implemented in #8728 and struct writing to ORC is covered by https://github.com/rapidsai/cudf/issues/7830 . As a result, I'm going to close this issue to consolidate discussion in 7830.
